### DISCUSSION TITLE
new ExceptionMapper behaviour since version 3.9

### DISCUSSION THREAD

**Discussion by iompo** [quarkusio/quarkus#39861] [Q&A]
I have an application that uses `quarkus-rest` and `quarkus-undertow`. Within the app I've defined a custom Exception Mapper:
```
@Provider
public class CustomExceptionMapper implements ExceptionMapper<Exception> {
    private static final Logger LOGGER = Logger.getLogger(CustomExceptionMapper.class.getName()); 
    
    @Override
    public Response toResponse(Exception exception) {
        LOGGER.log(Level.SEVERE, exception.getMessage(), exception);
        if (exception instanceof MyCustomException) {
            return Response.status(((MyCustomException) exception).getStatus()).entity(exception.getMessage()).build();
        }

        return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Something went wrong").build();
  }
}
```
After the update I can't access any of the static resources defined in the app (it is working fine with version 3.8.3). If I define it just for MyCustomException (e.g. `ExceptionMapper<MyCustomException>`) then everything works fine again. Is that intended? How can I replicate the current behavior I have?

Marco.

[the remark this answers is not in the document]
**Reply by geoand** [reply]
Let me check, I think there was something we did specifically to make that work...

[the remark this answers is not in the document]
**Reply by geoand** [reply]
I think it broke from 5219229529bbff2ff8a2322d78edac7ae639e197 - cc @phillip-kruger

[the remark this answers is not in the document]
**Reply by phillip-kruger** [reply]
Yes that would be my changes that broke this. @iompo do you have a small reproducer I can look at?

[the remark this answers is not in the document]
**Reply by iompo** [reply]
@phillip-kruger reproducer available [here](https://github.com/iompo/quarkus-static-res-fail), there's a failing test that runs fine if `CustomExceptionMapper` implements `ExceptionMapper<MyCustomException>` instead of `ExceptionMapper<Exception>`

[the remark this answers is not in the document]
**Reply by phillip-kruger** [reply]
Thanks. Looking now.

[the remark this answers is not in the document]
**Reply by phillip-kruger** [reply]
This will fix it. https://github.com/quarkusio/quarkus/pull/39897
